docs/admin/reference-architectures.md

```diff
@@ -0,0 +1,163 @@
+# Reference architectures
+
+This document provides prescriptive solutions and reference architectures to
+support successful deployments of up to 2000 users and outlines at a high-level
+the methodology currently used to scale-test Coder.
+
+## General concepts
+
+This section outlines core concepts and terminology essential for understanding
+Coder's architecture and deployment strategies.
+
+### Administrator
+
+An administrator is a user role within the Coder platform with elevated
+privileges. Admins have access to administrative functions such as user
+management, template definitions, insights, and deployment configuration.
+
+### Coder
+
+Coder, also known as _coderd_, is the main service recommended for deployment
+with multiple replicas to ensure high availability. It provides an API for
+managing workspaces and templates. Each _coderd_ replica has the capability to
+host multiple provisioners (`provisionerd`).
+
+### User
+
+A user is an individual who utilizes the Coder platform to develop, test, and
+deploy applications using workspaces. Users can select available templates to
+provision workspaces. They interact with Coder using the web interface, the CLI
+tool, or directly calling API methods.
+
+### Workspace
+
+A workspace refers to an isolated development environment where users can write,
+build, and run code. Workspaces are fully configurable and can be tailored to
+specific project requirements, providing developers with a consistent and
+efficient development environment. Workspaces can be autostarted and
+autostopped, enabling efficient resource management.
+
+Users can connect to workspaces using SSH or via workspace applications like
+`code-server`, facilitating collaboration and remote access. Additionally,
+workspaces can be parameterized, allowing users to customize settings and
+configurations based on their unique needs. Workspaces are instantiated using
+Coder templates and deployed on resources created by provisioners.
+
+### Template
+
+A template in Coder is a predefined configuration for creating workspaces.
+Templates streamline the process of workspace creation by providing
+pre-configured settings, tooling, and dependencies. They are built by template
+administrators on top of Terraform, allowing for efficient management of
+infrastructure resources. Additionally, templates can utilize Coder modules to
+leverage existing features shared with other templates, enhancing flexibility
+and consistency across deployments. Templates describe provisioning rules for
+infrastructure resources offered by cloud providers.
```

Comment:
s/cloud/Terraform/ ? I feel like we shouldn't constrain this statement to cloud providers only.

Reply:
Yeah, good call. _Terraform_ would fit better here.